.gitlab/scripts/run-benchmarks-with-profiler.sh

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env bash
+
+set -e
+
+CANDIDATE_SRC="/app/candidate/"
+BASELINE_SRC="/app/baseline/"
+
+mkdir -p "${ARTIFACTS_DIR}/candidate-profile"
+cd "$CANDIDATE_SRC/ddtrace/tracer/"
+
+PROFILER_RUNS_START_DATE=$(date -u --iso-8601=minutes | sed 's/\+.*/Z/')
+
+CANDIDATE_START_DATE=$(date -u --iso-8601=seconds | sed 's/\+.*/Z/')
+go test -run=XXX -bench "BenchmarkConcurrentTracing|BenchmarkStartSpan" \
+  -cpuprofile "${ARTIFACTS_DIR}/candidate-profile/cpu.pprof" \
+  -memprofile "${ARTIFACTS_DIR}/candidate-profile/delta-heap.pprof" \
+  -benchmem -count 10 -benchtime 2s ./...
+CANDIDATE_END_DATE=$(date -u --iso-8601=seconds | sed 's/\+.*/Z/')
+
+BASELINE_START_DATE=$(date -u --iso-8601=seconds | sed 's/\+.*/Z/')
+mkdir -p "${ARTIFACTS_DIR}/baseline-profile"
+cd "$BASELINE_SRC/ddtrace/tracer/"
+go test -run=XXX -bench "BenchmarkConcurrentTracing|BenchmarkStartSpan" \
+  -cpuprofile "${ARTIFACTS_DIR}/baseline-profile/cpu.pprof" \
+  -memprofile "${ARTIFACTS_DIR}/baseline-profile/delta-heap.pprof" \
+  -benchmem -count 10 -benchtime 2s ./...
```

Comment:
It seems like the list of benchmarks (and some of the options) are now duplicated in 4 places, (`run-benchmarks.sh` being the other file with 2 duplications).

Ideally that could be avoided (maybe via a shared bash include file). Alternatively some comment pointing out the duplications would be helpful.

Reply:
Ah, the duplication is now actually a problem. This just landed: https://github.com/DataDog/dd-trace-go/pull/1672/files

This PR will need adjustment.

Reply:
I will update it :+1: